mono/utils/mono-cgroup.c

```diff
@@ -0,0 +1,955 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*++
+
+Module Name:
+
+    mono-cgroup.c
+
+Abstract:
+    Read the memory limit for the current process
+
+    Adapted from runtime src/coreclr/gc/unix/cgroup.cpp 
+    - commit 28ec20194010c2a3d06f2217998cfcb8e8b8fb5e
+--*/
```

Comment:
let's wrap the whole source file in `HAVE_CGROUP_SUPPORT` when we have it (you'll need to use `MONO_EMPTY_SOURCE_FILE` at the end, see e.g. mono-endian.c)